tests/Command/GenerateCommandTest.php

```diff
@@ -0,0 +1,175 @@
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class GenerateCommandTest extends TestCase
+{
+    public function testExecute()
+    {
+        $commandTester = $this->buildCommand($this->mockContainer());
+        $commandTester->execute([
+            '--dest' => 'src',
+            'bundle' => ['SonataAcmeBundle'],
+        ]);
+
+        $this->assertContains(
+            'done!',
+            $commandTester->getDisplay()
+        );
+    }
+
+    public function testExecuteWrongDest()
+    {
+        $commandTester = $this->buildCommand($this->getMockBuilder(ContainerInterface::class)->getMock());
+
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage("The provided destination folder 'fakedest' does not exist!");
+
+        $commandTester->execute([
+            '--dest' => 'fakedest',
+        ]);
+    }
+
+    public function testNoArgument()
+    {
+        $commandTester = $this->buildCommand($this->mockContainerWithKernel());
+
+        $commandTester->execute([
+            '--dest' => 'src',
+        ]);
+
+        $this->assertContains(
+            'You must provide a bundle name!',
+            $commandTester->getDisplay()
+        );
+    }
+
+    public function testFakeBundleName()
+    {
+        $commandTester = $this->buildCommand($this->mockContainerWithKernel());
+
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage("The bundle 'FakeBundle' does not exist or not registered in the kernel!");
+
+        $commandTester->execute([
+            '--dest' => 'src',
+            'bundle' => ['FakeBundle'],
+        ]);
+
+        $this->assertContains(
+            'You must provide a bundle name!',
+            $commandTester->getDisplay()
+        );
+    }
+
+    public function testNotExtendableBundle()
+    {
+        $commandTester = $this->buildCommand($this->mockContainerWithKernel(new \Symfony\Bundle\NotExtendableBundle()));
+
+        $commandTester->execute([
+            '--dest' => 'src',
+            'bundle' => ['NotExtendableBundle'],
+        ]);
+
+        $this->assertContains(
+            sprintf('Ignoring bundle : "Symfony\Bundle\NotExtendableBundle"'),
+            $commandTester->getDisplay()
+        );
+    }
+
+    public function testInvalidFolderStructure()
+    {
+        $commandTester = $this->buildCommand(
+            $this->mockContainerWithKernel(new \Application\Sonata\NotExtendableBundle())
+        );
+
+        $commandTester->execute([
+            '--dest' => 'src',
+            'bundle' => ['NotExtendableBundle'],
+        ]);
+
+        $this->assertContains(
+            sprintf('Application\Sonata\NotExtendableBundle : wrong directory structure'),
+            $commandTester->getDisplay()
+        );
+    }
+
+    private function buildCommand($container)
```

Comment:
This could be simplified to a setUp function

Reply:
Not sure how as I need to set the container on command and I return command tester. I simplified it a little.

Reply:
No problem then :)